salt/auth/__init__.py

```diff
@@ -292,45 +292,50 @@ def authenticate_eauth(self, load):
     def authenticate_key(self, load, key):
         '''
         Authenticate a user by the key passed in load.
-        Return the effective user id (name) if it's differ from the specified one (for sudo).
-        If the effective user id is the same as passed one return True on success or False on
+        Return the effective user id (name) if it's different from the specified one (for sudo).
+        If the effective user id is the same as the passed one, return True on success or False on
         failure.
         '''
+        error_msg = 'Authentication failure of type "user" occurred.'
+        if 'key' not in load:
```

Comment:
So this is intended to see if the key exists whereas L305 seems to return False if the key does not exist or if it does exist and evals to False? Should those be combined?

Reply:
Oh, I was doing it this way to avoid a `KeyError` on the `pop` call, which I discovered in writing tests. I am probably being too cautious in trying not to disturb the code below. I could combine these two False returns into one with a `load.get('key')` instead.

Reply:
Yeah, this if block and the pop can be combined into:

```python
auth_key = load.pop('key', None)
if auth_key is None:
    log.warning(error_msg)
    return False
```

Reply:
Ah, I forgot you could add a default with `pop`. Thanks!